pandas/core/arrays/categorical.py

```diff
@@ -97,26 +98,23 @@ def func(self, other):
             else:
                 other_codes = other._codes
 
-            f = getattr(self._codes, opname)
-            ret = f(other_codes)
+            ret = op(self._codes, other_codes)
             mask = (self._codes == -1) | (other_codes == -1)
             if mask.any():
-                # In other series, the leads to False, so do that here too
-                if opname == "__ne__":
-                    ret[(self._codes == -1) & (other_codes == -1)] = True
-                else:
-                    ret[mask] = False
+                ret[mask] = fill_value
```

Comment:
Does it make sense to simplify further to only `ret[(self._codes == -1) | (other_codes == -1)] = fill_value` (or remove the `if mask.any()`)?

Reply:
I'm OK with it either way; we use the `if mask.any()` check elsewhere